climada/engine/supplychain.py

```diff
@@ -0,0 +1,342 @@
+                     'eeioa': self._eeioa_calc}
+
+        # Compute coefficients based on selected IO approach
+        coefficients = np.zeros_like(self.mriot_data, dtype=np.float32)
+        if io_approach in ['leontief', 'eeioa']:
+            for col_i, col in enumerate(self.mriot_data.T):
+                if self.total_prod[col_i] > 0:
+                    coefficients[:, col_i] = np.divide(col, self.total_prod[col_i])
+                else:
+                    coefficients[:, col_i] = 0
+        else:
+            for row_i, row in enumerate(self.mriot_data):
+                if self.total_prod[row_i] > 0:
+                    coefficients[row_i, :] = np.divide(row, self.total_prod[row_i])
+                else:
+                    coefficients[row_i, :] = 0
+
+        inverse = np.linalg.inv(np.identity(len(self.mriot_data)) - coefficients)
+        inverse = inverse.astype(np.float32)
+
+        # Calculate indirect impacts
+        self.indirect_impact = np.zeros_like(self.direct_impact, dtype=np.float32)
+        risk_structure = np.zeros(np.shape(self.mriot_data) + (len(self.years),),
+                                  dtype=np.float32)
+
+        # Loop over years indices:
+        for year_i, _ in enumerate(tqdm(self.years)):
+            direct_impact_yearly = self.direct_impact[year_i, :]
+
+            direct_intensity = np.zeros_like(direct_impact_yearly)
+            for idx, (impact, production) in enumerate(zip(direct_impact_yearly,
+                                                           self.total_prod)):
+                if production > 0:
+                    direct_intensity[idx] = impact/production
+                else:
+                    direct_intensity[idx] = 0
+
+            # Calculate risk structure based on selected IO approach
+            risk_structure = io_switch[io_approach](direct_intensity, inverse,
+                                                    risk_structure, year_i)
+            # Total indirect risk per sector/country-combination:
+            self.indirect_impact[year_i, :] = np.nansum(
+                risk_structure[:, :, year_i], axis=0)
+
+        self.indirect_aai_agg = self.indirect_impact.mean(axis=0)
+
+        self.io_data = {}
+        self.io_data.update({'coefficients': coefficients, 'inverse': inverse,
+                             'risk_structure' : risk_structure,
+                             'io_approach' : io_approach})
+
+    def calc_total_impact(self):
+        """Calculate total impacts summing direct and indirect impacts."""
+        self.total_impact = self.indirect_impact + self.direct_impact
+        self.total_aai_agg = self.total_impact.mean(axis=0)
+
+    def _map_exp_to_mriot(self, exp_regid, mriot_type):
+        """Map regions names in exposure into theiInput-output regions names."""
+
+        if mriot_type == 'WIOD':
+            mriot_reg_name = countries_by_numeric.get(str(exp_regid)).alpha3
```

Comment:
This line only works if `exp_regid` is in the format readable by `countries_by_numeric`.

Reply:
Yes, but that is what is supposed to be for a WIOD type of table, which has countries only. So one uses exposure at the country level.

Reply:
Yes, but I can use different standards for naming the countries. There is not only ISO3 strings. Or I could use a different country naming scheme.

Reply:
Is there such a funcionality in CLIMADA exposures, i.e., changing reg_id type? If yes, I am missing that and I will add the comment. If not, I see it as going back to WIOD-oriented (un-needed) recommendations